ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -236,14 +247,16 @@ private[ember] object Parser {
       private val lf: Byte = '\n'.toByte
 
       sealed trait ParsePreludeResult
-      final case class ParsePreludeError(
-          throwable: Throwable,
+      case class ParsePreludeError(
+          message: String,
+          caused: Option[Throwable],
           method: Option[Method],
           uri: Option[Uri],
           httpVersion: Option[HttpVersion]
-      ) extends Throwable(
+      ) extends Exception(
```

Comment:
Why is `message` not part of this constructor?  Would help with the stack trace.